plenoptic/simulate/models/naive.py

```diff
@@ -28,4 +38,180 @@ def forward(self, img):
+    """
+
+    def __init__(
+        self,
+        kernel_size: Union[int, Tuple[int, int]],
+        std: float = 3.0,
+        pad_mode: str = "circular",
+    ):
+        super().__init__()
+        assert std > 0, "Gaussian standard deviation must be positive"
+        self.std = nn.Parameter(torch.tensor(std))
+        self.kernel_size = kernel_size
+        self.pad_mode = pad_mode
+
+    @property
+    def filt(self):
+        filt = circular_gaussian2d(self.kernel_size, self.std)
+        return filt
+
+    def forward(self, x: Tensor) -> Tensor:
+        self.std.data = self.std.data.abs()  # ensure stdev is positive
+
+        x = same_padding(x, self.kernel_size, pad_mode=self.pad_mode)
+        y = F.conv2d(x, self.filt)
+        return y
+
+
+class CenterSurround(nn.Module):
+    """Center-Surround, Difference of Gaussians (DoG) filter model. Can be either
+    on-center/off-surround, or vice versa.
+
+    Filter is constructed as:
+    .. math::
+        f &= amplitude_ratio * center - surround \\
+        f &= f/f.sum()
+
+    The signs of center and surround are determined by `center` argument. The standard
+    deviation of the surround Gaussian is constrained to be at least `width_ratio_limit`
+    times that of the center Gaussian.
+
+    Parameters
+    ----------
+    kernel_size:
+        Shape of convolutional kernel.
+    on_center:
+        Dictates whether center is on or off; surround will be the opposite of center
+        (i.e. on-off or off-on).
+    width_ratio_limit:
+        Sets a lower bound on the ratio of `surround_std` over `center_std`.
+        The surround Gaussian must be wider than the center Gaussian in order to be a
+        proper Difference of Gaussians. `surround_std` will be clamped to `ratio_limit`
+        times `center_std`.
+    amplitude_ratio:
+        Ratio of center/surround amplitude. Applied before filter normalization.
+    center_std:
+        Standard deviation of circular Gaussian for center.
+    surround_std:
+        Standard deviation of circular Gaussian for surround. Must be at least
+        `ratio_limit` times `center_std`.
+    pad_mode:
+        Padding for convolution, defaults to "circular".
+    """
+
+    def __init__(
+        self,
+        kernel_size: Union[int, Tuple[int, int]],
+        on_center: bool = True,
+        width_ratio_limit: float = 4.0,
+        amplitude_ratio: float = 1.25,
+        center_std: float = 1.0,
+        surround_std: float = 4.0,
+        pad_mode: str = "circular",
+    ):
+        super().__init__()
+
+        assert width_ratio_limit > 1.0, "stdev of surround must be greater than center"
+        assert amplitude_ratio >= 1.0, "ratio of amplitudes must at least be 1."
+
+        self.on_center = on_center
+        self.kernel_size = kernel_size
+        self.width_ratio_limit = width_ratio_limit
+        self.register_buffer("amplitude_ratio", torch.tensor(amplitude_ratio))
```

Comment:
oh this `register_buffer` is neat. But why register only `amplitude_ratio` and not `width_ratio_limit`? (or others)

Reply:
`amplitude_ratio` determines how the center surround filters are normalized and need to be specified (e.g. for `frontend.OnOff`'s state_dict. The `width_ratio_limit` is only necessary for training.